src/main/java/thePackmaster/SpireAnniversary5Mod.java

```diff
@@ -743,35 +786,73 @@ public void update() {
     }
 
     @Override
-    public ArrayList<String> onSave() {
+    public SpireAnniversary5SaveWrapper onSave() {
+        SpireAnniversary5SaveWrapper wrapper = new SpireAnniversary5SaveWrapper();
         ArrayList<String> packIDs = new ArrayList<>();
         for (AbstractCardPack pack : currentPoolPacks) {
             packIDs.add(pack.packID);
         }
-        return packIDs;
+        wrapper.currentPoolPacks = packIDs;
+        wrapper.currentHat = Hats.currentHat;
+        return wrapper;
     }
 
     @Override
-    public void onLoad(ArrayList<String> strings) {
+    public void onLoad(SpireAnniversary5SaveWrapper saved) {
```

Comment:
I bet this will break runs in progress for anyone who started a run before this was changed and saves/recompiles/loads with the new code.

Things are still in development so it isn't that big a deal (people can always check out the older code and finish their run on it if they want), but it means that when we merge this we'll want to warn people. And that we wouldn't want to make a change like this after release (unlikely to be an issue since I doubt we'll make big changes like this after release).

Alternatively, the way to avoid this is to make the Hat into a separate CustomSavable -- you can have more than one, and it doesn't need to only be on the main mod class like it is here. For example, see https://github.com/modargo/corruptthespire/blob/master/src/main/java/corruptthespire/savables/SavableCorruption.java -- which also requires a line in `receivePostInitialize` to add the save field, `BaseMod.addSaveField(SavableCorruption.SaveKey, new SavableCorruption());`. Not really that much work and IMO keeps things better organized (instead of having the pack pool and hat lumped together).

Gk can also weigh in with his thoughts on whether it's worth the work to avoid breaking in progress runs.

Reply:
100% agree with this as well. Shouldn't have used the custom savable on the main mod file to begin with. Would prefer use BaseMod.addCustomSaveField

Reply:
OK, I figured a save wrapper may not have been the best option. I'll see about fixing this potentially based off the example; never used a CustomSavable outside of mod file contexts before